pagopa/io-app
# feat: [IOAPPCIT-38] Crypto key pair refactoring

## Short description
This PR makes a refactoring of crypto key pair management.

## List of changes proposed in this pull request
1. **package.json**: Links to  the new metadata specs.
1. **ts/boot/configureStoreAndPersistor.ts**: whitelist `lollipop` redux store config.
1. **ts/sagas/startup.ts**: track mixpanel key events for opted-in users.
1. **ts/sagas/startup/authenticationSaga.ts**: add key pair generation.
1. **ts/sagas/startup/__tests__/authenticationSaga.test.ts**: update the test to make it work.
1. **ts/sagas/startup/generateCryptoKeyPair.ts**: refactor key pair generation and event generation persistence.
1. **ts/store/actions/lollipop.ts**: `lollipop` redux store `actions`.
1. **ts/store/actions/types.ts**: add `lollipop` action to `action types`.
1. **ts/store/reducers/__mock__/backendStatus.ts**: add to the lollipop flag the new `min_app_version` field.
1. **ts/store/reducers/__tests__/lollipop.test.ts**: test the brand new `min_app_version` lollipop field.
1. **ts/store/reducers/backendStatus.ts**: add reducer to retrieve `min_app_version` of the lollipop remote flag.
1. **ts/store/reducers/index.ts**: add `lollipop` state to redux store.
1. **ts/store/reducers/types.ts**: add `lollipop` to global state reducers.
1. **ts/store/reducers/lollipop.ts**: add `lollipop` state reducer.
1. **ts/utils/__tests__/crypto.test.ts**: add test to key events information save/retrieval.
1. **ts/utils/crypto.ts**: utility function to save/retrieve key events information.

## How to test
`lollipop` in the backend api server is already enabled through the [`min_app_version` field](https://github.com/pagopa/io-dev-api-server/blob/42c26404cb3dfe68f6726b9f6027b24cb0699baf/src/payloads/backend.ts#L79). 
Check before login if a new `uuid` is set as `keyTag` in `LollipopState` on redux store. Try to log in and check if the key generation events are sent to mixpanel only if the user has opted-in to data tracking. On `MessageList` screen try to retrieve the public key and print it to the console to check the successful key generation and the right keyTag(redux store)/key(keystore/keychain) mapping.
